iree/tools/android/run_module_app/build_apk.sh

```diff
@@ -0,0 +1,267 @@
+  mkdir -p "${IREE_NATIVE_LIB_BUILD_DIR?}"
+fi
+
+echo ">>> Building IREE native libraries <<<"
+
+IREE_NATIVE_LIB_NAME=iree_tools_android_run_module_app_iree_run_module_app
+
+pushd "${IREE_NATIVE_LIB_BUILD_DIR?}"
+"${CMAKE_BIN?}" "${IREE_SOURCE_ROOT?}" -G Ninja \
+  -DCMAKE_BUILD_TYPE="${IREE_BUILD_TYPE?}" \
+  -DCMAKE_TOOLCHAIN_FILE="${ANDROID_NDK_ROOT?}/build/cmake/android.toolchain.cmake" \
+  -DANDROID_ABI="${IREE_ANDROID_ABI?}" \
+  -DANDROID_PLATFORM="android-${IREE_ANDROID_API_LEVEL?}" \
+  -DIREE_HOST_C_COMPILER="${CC_BIN?}" \
+  -DIREE_HOST_CXX_COMPILER="${CXX_BIN?}" \
+  -DIREE_BUILD_COMPILER=OFF \
+  -DIREE_BUILD_TESTS=OFF \
+  -DIREE_BUILD_SAMPLES=OFF
+"${NINJA_BIN?}" "${IREE_NATIVE_LIB_NAME?}"
+popd
+
+#####################
+# Package Android app
+#####################
+
+# Clean artifacts from previous runs.
+rm -rf "${IREE_ARTIFACT_ROOT?}/${IREE_APK_NAME?}.apk" "${IREE_APK_PARTS_DIR?}"
+
+echo ">>> Generating AndroidManifest.xml <<<"
+
+# Create an AndroidManifest.xml with proper target SDK version.
+mkdir -p "${IREE_APK_PARTS_DIR?}"
+IREE_ANDROID_API_LEVEL="${IREE_ANDROID_API_LEVEL?}" envsubst \
+  < "${IREE_NATIVE_APP_SOURCE_ROOT?}/AndroidManifest.xml.template" \
+  > "${IREE_APK_PARTS_DIR?}/AndroidManifest.xml"
+
+echo ">>> Preparing shared libraries and vm module information <<<"
+
+# Find the compiled iree_run_module_app shared library and symlink it to a
+# known location for packaging.
+mkdir -p "${IREE_NATIVE_LIB_DIR?}"
+IREE_NATIVE_LIB=$(find ${IREE_NATIVE_LIB_BUILD_DIR?} -name "lib${IREE_NATIVE_LIB_NAME?}.so")
+# Note: the target link name must match with
+# run_module_app/AndroidManifest.xml.template.
+ln -sf "${IREE_NATIVE_LIB?}" "${IREE_NATIVE_LIB_DIR?}/libiree_run_module_app.so"
+
+# Copy the VM FlatBuffer and iree-run-module invocation related information
+# over as assets.
+mkdir -p "${IREE_ASSET_DIR?}"
+# Note: the following files must match with run_module_app/src/main.cc.
+cp "${IREE_INPUT_MODULE_FILE?}" "${IREE_ASSET_DIR?}/module.vmfb"
+cp "${IREE_INPUT_BUFFER_FILE?}" "${IREE_ASSET_DIR?}/inputs.txt"
+echo -n "${IREE_ENTRY_FUNCTION?}" > "${IREE_ASSET_DIR?}/entry_function.txt"
+echo -n "${IREE_DRIVER?}" > "${IREE_ASSET_DIR?}/driver.txt"
+
+echo ">>> Compiling app resources <<<"
+
+# Generate the R.java for resources.
+mkdir -p "${IREE_RESOURCE_GEN_DIR?}"
+${AAPT_PACK?} --non-constant-id -m \
+  -M "${IREE_APK_PARTS_DIR?}/AndroidManifest.xml" \
+  -S "${IREE_NATIVE_APP_SOURCE_ROOT?}/res" \
+  -J "${IREE_RESOURCE_GEN_DIR?}" \
+  --generate-dependencies
+
+# Compile the R.java and create classes.dex out of it for Android.
+echo "Using javac: '${JAVAC_BIN}'"
+${JAVAC?} "${IREE_RESOURCE_GEN_DIR?}"/com/google/iree/run_module/*.java
+${DX?} --output="${IREE_APK_PARTS_DIR?}/classes.dex" "${IREE_APK_PARTS_DIR?}"
+
+echo ">>> Packaging apk file <<<"
+
+# Package assets and shared libraries into an apk file.
+${AAPT_PACK?} -m \
+  -M "${IREE_APK_PARTS_DIR?}/AndroidManifest.xml" \
```

Comment:
These don't by any chance have long names, do they?

Reply:
Not sure what do you mean here? Normally it's just something like `/path/to/iree/build-apk/parts/AndroidManifest.xml`. So shouldn't be too long.

Reply:
I mean the command line options. Rather than `-M`, maybe there's `--manifest` or something. I think it's better to use long names in scripts and such because they're more self explanatory and you're not typing it a bunch.

Reply:
I see. But yeah, unfortunately there is no longer version for `-m`, `-M`, `-J` etc.